.github/actions/zenodo-upload/zenodo_upload.py

```
@@ -0,0 +1,101 @@
+    # Publish
+    response = requests.post(f"{base_url}/{new_deposition_id}/actions/publish", headers=headers)
+    if response.status_code != 202:
+        print(f"Error publishing: {response.text}")
+        sys.exit(1)
+
+    doi = response.json()["doi"]
+    with open(os.environ["GITHUB_OUTPUT"], "a") as f:
+        f.write(f"doi={doi}\n")
+
+if __name__ == "__main__":
+    main()
```

Ruff (0.8.2)

No newline at end of file

Add trailing newline

(W292)